src/test/java/org/pesmypetcare/webservice/dao/KcalDaoTest.java

```diff
@@ -0,0 +1,381 @@
+            given(kcalsRef.get()).willReturn(futureQuery);
+            willThrow(ExecutionException.class).given(futureQuery).get();
+
+            kcalDao.getAllKcal(owner, petName);
+        }, EXCECUTION_EXC_MSG);
+    }
+
+    @Test
+    public void shouldReturnAllKcalsBetweenDatesOnDatabaseWhenRequested() throws DatabaseAccessException,
+        ExecutionException, InterruptedException {
+        given(db.collection(anyString())).willReturn(usersRef);
+        given(usersRef.document(anyString())).willReturn(ownerRef);
+        given(ownerRef.collection(anyString())).willReturn(petsRef);
+        given(petsRef.document(anyString())).willReturn(petRef);
+        given(petRef.collection(anyString())).willReturn(kcalsRef);
+        given(kcalsRef.get()).willReturn(futureQuery);
+        given(futureQuery.get()).willReturn(querySnapshot);
+        given(querySnapshot.getDocuments()).willReturn(kcalsDocuments);
+        given(kcalsDocuments.iterator()).willReturn(it);
+        given(it.hasNext()).willReturn(true);
+        given(it.hasNext()).willReturn(true);
+        given(it.hasNext()).willReturn(false);
+
+        List<Map<String, Object>> list = kcalDao.getAllKcalsBetween(owner, petName, date, date2);
+
+        assertEquals(kcalList, list, "Should return a List containing all meals between two dates");
+    }
+
+    @Test
+    public void shouldThrowDatabaseAccessExceptionWhenGetAllKcalsBetweenFromDatabaseReceivesInterruptedException() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            given(db.collection(anyString())).willReturn(usersRef);
+            given(usersRef.document(anyString())).willReturn(ownerRef);
+            given(ownerRef.collection(anyString())).willReturn(petsRef);
+            given(petsRef.document(anyString())).willReturn(petRef);
+            given(petRef.collection(anyString())).willReturn(kcalsRef);
+            given(kcalsRef.get()).willReturn(futureQuery);
+            willThrow(InterruptedException.class).given(futureQuery).get();
+
+            kcalDao.getAllKcalsBetween(owner, petName, date, date2);
+        }, INTERRUPTED_EXC_MSG);
+    }
+
+    @Test
+    public void shouldThrowDatabaseAccessExceptionWhenGetAllKcalsBetweenFromDatabaseReceivesExecutionException() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            given(db.collection(anyString())).willReturn(usersRef);
+            given(usersRef.document(anyString())).willReturn(ownerRef);
+            given(ownerRef.collection(anyString())).willReturn(petsRef);
+            given(petsRef.document(anyString())).willReturn(petRef);
+            given(petRef.collection(anyString())).willReturn(kcalsRef);
+            given(kcalsRef.get()).willReturn(futureQuery);
+            willThrow(ExecutionException.class).given(futureQuery).get();
+
+            kcalDao.getAllKcalsBetween(owner, petName, date, date2);
+        }, EXCECUTION_EXC_MSG);
+    }
+
+    @Test
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [NCSS for this method is 16 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5400494)